react1/main.js

```diff
@@ -0,0 +1,124 @@
+console.log("ЗАДАНИЕ 1")
+
+function loop (times = 0, callback = null) {
+    for (let i = 0; i < times; i++) {
+        console.log( callback )
+    }
+}
+
+loop(4)
+
+console.log("ЗАДАНИЕ 2")
+
+function calculateArea(height, width = 0) {
+    if (width == 0) {
+        return {
+            area: `Площадь ${height * height}`,
+            figure: 'Квадрат',
+            input: `Сторона квадрата ${height}`
+        }
+    } else {
+        return {
+            area: `Площадь ${height * width}`,
+            figure: 'Прямоугольник',
+            input: `Высота ${height}, ширина ${width}`
+        }
+    }
+}
+console.log(calculateArea(2))
+console.log(calculateArea(2,4))
+
+console.log("ЗАДАНИЕ 3")
+
+class Human {
+    constructor (name, age, dateOfBirth) {
+        this.name = name;
+        this.age = age;
+        this.dateOfBirth = dateOfBirth;
+    }
+
+    displayInfo () {
+        return `Имя: ${this.name}, возраст: ${this.age}, день рождение: ${this.dateOfBirth}`;
+    }
+}
+
+
+class Employee extends Human {
+    constructor (name, age, dateOfBirth, salaru, depatment){
+        super(name, age, dateOfBirth);
+        this.salary = salaru;
+        this.depatment = depatment;
+    }
+
+    displayInfo (){
+        return super.displayInfo() + `, зарплата: ${this.salary}, отдел: ${this.depatment}`;
+    }
+}
+
+
+class Manager extends Employee{
+    constructor(name, age, dateOfBirth, salaru, depatment){
+        super(name, age, dateOfBirth, salaru, depatment);
+        this.developerArr = [];
+    }
+
+    addDeveloper(developer){
+        this.developerArr.push(developer);
+        developer.manager = this;
+    }
+
+    deleteDeveloper(developer){
+        this.developerArr.splice(this.developerArr.indexOf(developer),1);
+        developer.manager = '';
+    }
+}
+
+class Developer extends Employee{
+    constructor(name, age, dateOfBirth, salaru, depatment){
+        super(name, age, dateOfBirth, salaru, depatment);
+        this.manager = '';
+    }
+    removeManager(manager){
```

Comment:
```suggestion
    changeManager(manager){
```
больше похоже, что мы его меняем:)

Reply:
ну я что-то даже не подумал что так написать можно было)))))